packages/medusa/src/api/admin/promotions/validators.ts

```diff
@@ -155,51 +155,67 @@ const promoRefinement = (promo) => {
   return true
 }
 
-export type AdminCreatePromotionType = z.infer<typeof _AdminCreatePromotion>
-const _AdminCreatePromotion = z
+export type AdminCreatePromotionType = z.infer<typeof CreatePromotion>
+export const CreatePromotion = z
   .object({
     code: z.string(),
     is_automatic: z.boolean().optional(),
     type: z.nativeEnum(PromotionType),
     campaign_id: z.string().nullish(),
-    campaign: AdminCreateCampaign.optional(),
+    campaign: CreateCampaign.optional(),
     application_method: AdminCreateApplicationMethod,
     rules: z.array(AdminCreatePromotionRule).optional(),
   })
   .strict()
 
-export const AdminCreatePromotion = (customSchema?: ZodObject<any, any>) => {
-  const schema = customSchema
-    ? _AdminCreatePromotion.merge(customSchema)
-    : _AdminCreatePromotion
+export const AdminCreatePromotion = (
+  additionalDataValidator?: ZodObject<any, any>
+) => {
+  if (!additionalDataValidator) {
```

Comment:
q: are you doing this for all validators for every hook you create? 

Maybe worth adding a helper if its common:

```
export const AdminCreatePromotion = wrapAdditionalDataValidator(CreatePromotion)
```

Reply:
Nice suggestion. It did felt like additional work, but I kind of tuned myself to do it. But I will create a helper now. :)